settings.local.php configuration override

shellshocked59 opened the pull request:
I'd like to enable support for overriding settings.php with a settings.local.php on boxes.  This type of overriding is something I've always liked for enabling a non-version-controlled file to hold database and other configuration changes for local environments. 

You can see a longer discussion on this feature here: https://www.drupal.org/node/1118520.  Alternatively if you have a preferred alternative to this solution for local configuration I could use that.

pirog commented:
this seems pretty reasonable to me. only small comment would be do you think it should `settings.local.php` or `local.settings.php`? mention this only because we have a precedent with `default.settings.php`

shellshocked59 commented:
Sounds good, I updated the branch to include the new filename. I also added a default.local.settings.php file that explains the purpose of the file, and can be copied for developers to create their own local.settings.php files from a template.

Do I need to submit a new pull request to get the new commit in?

pirog commented:
Oh nice! Good call on the default file.

You can just push more commits to this branch and they'll get added to this PR.

pirog commented:
which it looks like you already did! @reynoldsalec can take it from here!

reynoldsalec commented:
Looks good to me on my Kalabox, merging in!
